### PR TITLE
NSAttributedString: Align methods with Darwin.

### PR DESCRIPTION
**Pull request by spevans**
- Add struct NSAttributedStringKey and use for the attribute key
  instead of String.

### CODE REVIEW

**Comment by spevans** [comment]
@swift-ci please test

**Comment by spevans** [comment]
@swift-ci please test

File: Foundation/NSAttributedString.swift [hunk 1]
```diff
@@ -42,73 +58,87 @@ open class NSAttributedString: NSObject, NSCopying, NSMutableCopying, NSSecureCo
...
+
+    /// The character contents of the receiver as an NSString object.
     open var string: String {
         return _string._swiftObject
     }
-    
-    open func attributes(at location: Int, effectiveRange range: NSRangePointer) -> [String : Any] {
+
+    /// Returns the attributes for the character at a given index.
+    open func attributes(at location: Int, effectiveRange range: NSRangePointer) -> [NSAttributedStringKey : Any] {
```
Comment (ianpartridge) [inline Foundation/NSAttributedString.swift]: The `NSRangePointer` should be optional: https://developer.apple.com/documentation/foundation/nsattributedstring/1415682-attributes

File: Foundation/NSAttributedString.swift [hunk 2]
```diff
@@ -241,14 +272,18 @@ private extension NSAttributedString {
...
         guard _string.length > 0 else {
             return
         }
         
         let range = CFRange(location: 0, length: _string.length)
         if let attrs = attrs {
-            CFRunArrayInsert(_attributeArray, range, attrs._cfObject)
+            var _attrs: [String: Any] = [:]
+            for (key, value) in attrs {
+                _attrs[key.rawValue] = value
```
Comment (ianpartridge) [inline Foundation/NSAttributedString.swift]: Bleh.

File: Foundation/NSAttributedString.swift (hunk 1 above)
Comment (spevans) [reply Foundation/NSAttributedString.swift]: fixed

**Comment by spevans** [comment]
@swift-ci please test

**Comment by spevans** [comment]
@swift-ci please test

**Comment by ianpartridge** [comment]
@swift-ci please test and merge

**Comment by ianpartridge** [comment]
@swift-ci please test and merge

**Pull request merged**